File: legacy_iceweb/code/DAILY_SPECTROGRAMS/create_UTday_spectrograms.pl
#!/usr/bin/perl
# Glenn Thompson, September 1999
# This script produces 24 hour spectrograms

# Setup paths & modules
$ICEWEB=$ENV{HOME}; # iceweb home
$PFS="$ENV{PFS}"; # parameter files

use lib "$ENV{ANTELOPE}/data/perl"; # antelope-perl interface
use Datascope; # Datascope.pm module

use lib "$ENV{HOME}/ICEWEB_UTILITIES"; # utilities for talking to iceweb parameter files
use iceweb_perl_utilities qw(send_IceWeb_error read_volcanoes get_time);
$SPEC_GIFS=pfget("$PFS/paths.pf","SPEC_GIFS");

# Get days_ago
$days_ago=$ARGV[0];

# Run Matlab code - this puts postscript images into /tmp directory
$|=1;
system("$ICEWEB/ICEWEB_UTILITIES/start_matlab_engine create_UTday_spectrograms $ICEWEB/DAILY_SPECTROGRAMS $days_ago");

# Read list of volcanoes
@volcanoes=read_volcanoes();
$NUMVOLCANOES=$#volcanoes;
print "@volcanoes\n";

# LOOP OVER ALL VOLCANOES IN ICEWEB PARAMETER FILE
for ($volcano_num=0;$volcano_num<=$NUMVOLCANOES;$volcano_num++) {
	$volcano = $volcanoes[$volcano_num];

	# if a 24 hr spectrogram has been made for this volcano
	($year,$mon,$mday,$hour,$min)=get_time("ut",$days_ago);
	$datestr="$year" . "$mon" . "$mday";
print "$datestr\n";
	$tempfile="/tmp/$volcano" . "_$datestr" . ".ps";
	if (-e $tempfile) {
		
		print sprintf("\n Converting 24 hr spectrogram for %s\n",$volcano);

		# convert temporary ps to permanent gif file
		$relname = "$year" . "$mon" . "$mday" . "_" . "$hour" . "$min";
		$TEMPPATH="$SPEC_GIFS/$volcano";
		$PERMPATH="$TEMPPATH/$year" . "$mon";
		unless (-e $PERMPATH) {system("mkdir $PERMPATH")};
		$permname="$PERMPATH/$volcano_$datestr";
		system("alchemy $tempfile -Zm2 -Zc1 -Zo 600p -Z+ $permname.gif -go -Q");
		system("rm $tempfile");

		# make copy if today or yesterday (program will run for ANY day though!)
		if ($days_ago==0) {system("cp $permname.gif $TEMPPATH/currentday.gif")}
		elsif ($days_ago==1) {system("cp $permname.gif $TEMPPATH/lastday.gif")};
		} 
	else
	{
		print sprintf("\n No 24 hr spectrogram for %s\n",$volcano);
	};

};
 

File: legacy_iceweb/code/DAILY_SPECTROGRAMS/create_UTday_spectrograms.m
function create_UTday_spectrograms(days_ago);
% spec data is saved in day-long files for each station
% This function plots day-long spectrograms for each volcano
% listed in the IceWeb parameter file

% add Antelope extensions
%use_antelope;

% global variables
global ICEWEB parameterspf PARAMETER_FILES TRUE FALSE;

% set iceweb home
ICEWEB='/home/iceweb';

% path for parameter files
PFS=[ICEWEB,'/PARAMETER_FILES'];

% create pointer to main parameter file
parameterspf=dbpf([PFS,'/parameters']);

% add paths to code
path(path,[ICEWEB,'/ICEWEB_UTILITIES']);

% add important variables
F=0.5:0.1:15.0;  % this is frequency range of archived spec data
TRUE=1;FALSE=0; % boolean

% make sure days_ago is defined - if it was missed, user probably wants todays data
if ~exist('days_ago','var')
	days_ago=0;
end

% read set of stations for each volcano from controlfile
volcanoes=read_iceweb_volcanoes;
numvolcanoes=length(volcanoes);

% work out UT year, month & day corresponding to 'days_ago'
dnum=epoch2dnum(0)-days_ago; % dnum for current UT time
[yr,mn,dy]=yyyymmdd(dnum);

% loop for all volcanoes in the controlfile
for volcano_num=1:numvolcanoes

	% close any previous plots
	close all; 

	% define useful variables
	volcano=volcanoes{volcano_num}
	stations=read_iceweb_stations(volcano)
	numstations=length(stations);

	% fetch spec data for each station listed
	% for this volcano in the IceWeb parameter file for this UT day
	for frame_num=1:numstations
		% define useful variables
		station_num=numstations+1-frame_num;
		station=stations{station_num};

		% load spectral data for this day
		[data,DATA_FOUND]=load_spec_data(yr,mn,dy,station);

		if DATA_FOUND == TRUE
			T=data(:,1);
			A=data(:,2:147);
			% calculate position of spectrogram & trace data frames for this station
			[spectrogram_position,trace_position]=...
			calculate_frame_positions(numstations,frame_num,0.9);
			% plot data
			plot_spectrogram(A',F,T,station,frame_num,spectrogram_position,[],['']);
		end
		clear data;
	end


	if sum(DATA_FOUND) > 0  % if data was loaded for ANY station, do the following
		tstr=sprintf('%s %s',volcano,datestr(dnum,1));
		title(tstr,'Color',[0 0 0],'FontSize',[16], 'FontWeight',['bold']');
		add_colorbar;
		orient tall;
		psfile=['/tmp/',volcano,'_',yr,mn,dy,'.ps'];
		eval(['print -dpsc ',psfile]);
	else
		send_IceWeb_error(['no data to plot for ',volcano]);
	end	
end
